### DISCUSSION TITLE
Ability to inject JS into <head>

### DISCUSSION THREAD

**Discussion by benwoodward** [roam-garden/i#50] [Feature requests & Ideas]
I'd like to be able to add JS to a garden, e.g. Google Analytics, ConvertKit

**Comment by Stvad** [comment]
Definitely something I'm looking to build! How would you imagine UX working? 

- An entry field during upload where you can post a bunch of script tags?
- Something read from a selected Roam page?
- Something else?

**Reply by benwoodward** [reply]
To me it makes sense that Roam pages are just for content, and the website is where settings can be modified. 

I think Ghost has a good UI for this:

![image](https://user-images.githubusercontent.com/1472981/112788681-e4faf900-90b7-11eb-86d3-9662545e0b37.png)

**Reply by halilakin** [reply]
Curious if there has been any updates here.. I am looking forward to this feature as well.

**Reply by Stvad** [reply]
This is high on my priority list, I plan to look at it soon - probably after https://github.com/roam-garden/i/issues/67

**Reply by ybalkind** [reply]
Plus 1 for this. In my case I want it for adding hypothsis.is script so as to allow comments and annotation.

**Reply by Stvad** [reply]
ah yes I was thinking about integrating hypothes.is by default! but it does not seem to work well with the sliding notes structure =\

**Reply by ybalkind** [reply]
Yeah I feared that. My guess is that it attributes comments based on URL. So you and I might comment on the same post but we might see it on a different URL depending on sequence of sliding notes that we followed. It would only work well if we use the actual permalink of the post in question and then others would only see the comments if they are on the specific permalink as well.

I'm not a programmer so I could be wrong about all of this.

**Reply by Stvad** [reply]
that corresponds to my model of what is happening

**Reply by halilakin** [reply]
Do we have any updates on this?

**Reply by Stvad** [reply]
Apologies, life caught up with me a bit, so experiencing delays on this. It's still one of my next priorities for Roam Garden!

**Reply by halilakin** [reply]
Do we have any timeline on this though? Or any other workarounds?

**Reply by Stvad** [reply]
Don't have a specific timeline, but started to looking into this and expect to spend more time on it starting next Tue

**Reply by Stvad** [reply]
for workarounds: you can right now add arbitrary JS on a specific page, as any valid JSX syntax is supported #51

**Reply by Stvad** [reply]
made some progress, have a prototype of applying the additional metadata. working on delivery from the user

**Reply by halilakin** [reply]
Thanks so much @Stvad! Looking forward to this!
